cerbero/build/cookbook.py

```diff
@@ -467,7 +481,8 @@ def _load_recipes(self):
             # allow safe relocation of the recipes.
             if recipe.__file__ != st.filepath:
                 st.filepath = recipe.__file__
-            recipe.run_func_depending_on_built_version(async_tasks, self._reset_recipe_if_needed, recipe)
+            if not self._recipe_is_reset(recipe.name):
+                recipe.run_func_depending_on_built_version(async_tasks, self._reset_recipe_if_needed, recipe)
```

Comment:
The method it's calling already checks if self._recipe_is_reset(recipe.name), it's a bit confusing

Reply:
Actually, it looks weird but it is indeed needed here. If we don't use it, in case the `built_version` is an async task (such as in Git sources), a Future task will be scheduled in order to collect it. That will eventually call `git ls-remote` when it's unnecessary, since we already know the recipe is reset and we don't need to reset it because there was a change in the built_version

Reply:
That is strange, why would it call it? The function it's deferring is going to check if the recipe is reset  at the very beginning and exit

Reply:
Previously to calling the method that resets the recipe, you need to already have the proper `built_version` for a given recipe. In case the method to get that is an async method (which in Git it is), you can't do that sequentially. You need to schedule all the tasks and then gather the results. After that, you can call the reset method.

The async built_version command was done to make sure both fridge and the recipes use the proper commit for the names, resolving them by running `git ls-remote` when the repo has not been cloned.